datafusion/physical-plan/src/windows/mod.rs

```diff
@@ -257,22 +266,17 @@ fn create_built_in_window_expr(
                 return exec_err!("NTILE requires a positive integer, but finds NULL");
             }
 
-            if n.is_unsigned() {
-                let n: u64 = n.try_into()?;
-                Arc::new(Ntile::new(name, n, out_data_type))
-            } else {
-                let n: i64 = n.try_into()?;
-                if n <= 0 {
-                    return exec_err!("NTILE requires a positive integer");
-                }
-                Arc::new(Ntile::new(name, n as u64, out_data_type))
+            let n: i64 = get_integer(n)?;
+            if n <= 0 {
+                return exec_err!("NTILE requires a positive integer");
             }
+            Arc::new(Ntile::new(name, n as u64, out_data_type))
```

Comment:
removing the `n.is_unsigned` branch should be fine I think assuming people don't use ntile with numbers greater than what fits into a signed i64..